src/wp-includes/rest-api.php

```diff
@@ -1050,13 +1050,36 @@ function rest_cookie_collect_status() {
  * @since 5.6.0
  *
  * @global WP_User|WP_Error|null $wp_rest_application_password_status
+ * @global string|null $wp_rest_application_password_uuid
  *
  * @param WP_Error $user_or_error The authenticated user or error instance.
+ * @param array    $app_password  The Application Password used to authenticate.
  */
-function rest_application_password_collect_status( $user_or_error ) {
-	global $wp_rest_application_password_status;
+function rest_application_password_collect_status( $user_or_error, array $app_password = array() ) {
+	global $wp_rest_application_password_status, $wp_rest_application_password_uuid;
 
 	$wp_rest_application_password_status = $user_or_error;
+
+	if ( $app_password ) {
+		$wp_rest_application_password_uuid = $app_password['uuid'];
+	} else {
+		$wp_rest_application_password_uuid = null;
+	}
```

Comment:
Logic check that `uuid` is set in the array.
```suggestion
	if ( empty( $app_password['uuid'] ) ) {
		$wp_rest_application_password_uuid = null;
	} else {
		$wp_rest_application_password_uuid = $app_password['uuid'];
	}
```